upgrade.php

```diff
@@ -169,4 +169,20 @@
     exit();
 }
 
+echo 'Adding opens field to newsletter&hellip;<br />'.PHP_EOL;
+$result = $mysqli->query("ALTER TABLE `newsletters` ADD `opens` INT( 8 ) NULL DEFAULT 0 AFTER `distributed`;");
+if (!$result) {
+    if (mysqli_errno($mysqli) == 1060) {
+        echo 'Field already has been added<br />'.PHP_EOL;
+    }
+}
+
+echo 'Add new table newsletter_open for tracking&hellip<br />'.PHP_EOL;
+$result = $mysqli->query("CREATE TABLE newsletter_open (ip VARCHAR(64), newsletter_id INT(10));");
```

Comment:
No primary key to establish an index?

If the two combined are a primary key, do you not want to count the number of times an IP address has opened an email?